dhis-2/dhis-services/dhis-service-event-hook/src/main/java/org/hisp/dhis/eventhook/handlers/WebhookHandler.java

```diff
@@ -74,7 +78,11 @@ public void run(EventHook eventHook, Event event, String payload) {
 
     MultiValueMap<String, String> queryParams = new LinkedMultiValueMap<>();
     if (webhookTarget.getAuth() != null) {
-      webhookTarget.getAuth().apply(httpHeaders, queryParams);
+      try {
+        webhookTarget.getAuth().apply(applicationContext, httpHeaders, queryParams);
+      } catch (Exception e) {
+        throw new RuntimeException(e);
```

Comment:
I think this is the async exception we talked about. Maybe we should leave a comment stating that this is async and therefore the type of exception this is re-thrown as does not matter much which is why it just converts it to a runtime exception?